lua/luasnip/extras/init.lua

```diff
@@ -28,9 +28,46 @@ end
 
 setmetatable(lambda, LM)
 
+local function to_function(val, use_re)
+	if type(val) == "function" then
+		return val
+	end
+	if type(val) == "string" and not use_re then
+		return function()
+			return val
+		end
+	end
+	if type(val) == "string" and use_re then
+		return function(arg)
+			return arg:match(val)
+		end
+	end
+	if lambda.isPE(val) then
+		return lambda.instantiate(val)
+	end
+	assert(false, "Can't convert argument to function")
+end
+
+local function match(index, _then, _match, _else)
+	assert(_then, "You have to pass at least 2 arguments")
+	_then = to_function(_then)
+	_match = to_function(_match or "^$", true)
+	_else = to_function(_else or "")
+	local function func(arg)
+		if _match(arg[1][1]) then
```

Comment:
Allowing `match` to also be a lambda is a great idea, but then it'd be even better to pass the argument-nodes' text the same way as in the original lambda nodes (eg. '\n'-concat text from each node and pass these strings to `_match`), that way multiple nodes could be used in `_match`(shouldn't interrupt usage with simple patterns, I think).
I just realized that that should probably be done for `_then` and `_else` as well, to keep lambda-usage more uniform.

Reply:
Sure, I was a bit bussy today.

Reply:
I did it for a single node, is not clear to me how to do that for several nodes, like, should we do if all the matches work? Or a single match? I guess that will add complexity to the api and at that point better to use a normal  function node IMHO.

Reply:
I was thinking of only matching on the first node, but you're right, there would be some ambiguity about what 'true' means if several nodes are given in `index` (eg. only the first node matches, or all nodes match).
On the other hand, adding it would enable us to insert text when eg. two nodes have the same text (`m({1,2}, "equal!!", l._1:match(_2))`), which could be nice.

Upon thinking, that wouldn't be common at all, I can't think of a single snippet where it would make sense to have such functionality, nevermind that.

Reply:
Actually, in that case, could you add an `assert()` to make sure that the user is actually passing a single index and not a table?

Reply:
I feel like many of these functions accept a table as indices and we should probably be more explicit about the type of the parameters (I'll add those to some other functions too, as soon as I have some time).